frontegg/frontegg-angular
# FR-12554 - support-angular-16-NOT-backward-compatible

### Support angular 16
- upgrade angular version to 16 to support incompatible ivy bug 
- upgrade other packages to be compatible
- upgrading `ng-packagr` requires moving `umdModuleIds` from `ng.package.json` as it is not supported from v11
- also build the package differently as `defaultProject` is not supported in angular.json
- create new signals functions wrapping all of our observables with `toSignal`

### Example usage of signals 

```
app.component.ts
export class AppComponent {
  title = 'angular-16-demo';
  constructor(private fronteggAppService: FronteggAppService) { }
  isLoading = this.fronteggAppService.isLoadingSignal;
}
```

```
app.component.html
<div *ngIf="!isLoading()">
  <router-outlet></router-outlet>
</div>
```

### Gaps
- Figure out why we need `FronteggComponent` inside `entryComponents`(works fine with out it), because angular removed support for `entryComponents` in v16
- There are warnings appearing in the console because we use libs that is build as CommonJS (checked and the warning appear before but now there are 2 more packages)

### How should we test it
- running the new version with angular 14/15 to checkout if there is anything breaking
- running the new version with angular 16 to checkout everything is working as expected with old methods (observables)
- using the new signals inside the app and checkout that they are working as expected

### Releasing
- In my opinion this version should be a major `5 => 6` just like angular version `15=>16` , this way it will make it very clear to our customers that this version is the one compatible with angular 16 when upgrading their angular version
